src/main/app/form/validation.ts

```diff
@@ -106,8 +108,29 @@ export const isPhoneNoValid: Validator = value => {
   }
 };
 
+export const isEmailValid: Validator = value => {
+  if (!isValidEmail(value as string)) {
+    return 'invalid';
+  }
+};
+
 export const isFieldLetters: Validator = value => {
   if (!(value as string).match(/^[a-zA-Z][a-zA-Z\s]*$/)) {
     return 'invalid';
   }
 };
+
+export const doesNotKnowEmail = (value: Partial<Case>): string | void => {
+  if (value.respondentEmailAddress && value.doNotKnowRespondentEmailAddress === Checkbox.Checked) {
+    return 'incorrect';
+  }
+};
+
+export const isEitherFieldsFilledIn = (value: Partial<Case>): string | void => {
+  if (
+    isFieldFilledIn(value.respondentEmailAddress) === 'required' &&
+    value.doNotKnowRespondentEmailAddress !== Checkbox.Checked
```

Comment:
Perhaps this method should be renamed to something like `isRespondentEmailFilledOrNotKnown` as `isEitherFieldsFilledIn` doesn't match what this is doing and sounds like it would take an array of fields to check.

Reply:
On second thoughts it might be better to move this out of the shared `validation.ts` file and into the `content.ts` it belongs to since it's not really sharable to any other question.

Reply:
Ok cool, is it ok to still be tested inside validation.test or should I make a new file inside test/steps where we can test custom validation (assuming we'll probably have more in the future)?

Reply:
That's a good point. For the some of the more basic `content.ts` we've been relying on the functional tests to check that their working correctly, so if there's a functional test that covers this behaviour, then it may not be needed otherwise you can make `content.test.ts` next to it, to cover it.

The only downside with the functional tests is that the code coverage may go down even though technically there could be functional test that is validating this. The current best practices seem to suggest to [write more integration tests rather than unit tests](https://kentcdodds.com/blog/write-tests).

Reply:
in this context functional tests would be e2e tests on his pyramid. We don't really have integration tests

Reply:
Looks like the functional tests cover this behaviour, so I'll see what the coverage looks like and if its too low I'll add in the unit test